### PR TITLE
Error should be raised when we try to set slider of output variable [#19]

### PR DESCRIPTION
**Pull request by japamat0**
**[issue #19] 
the problem:**

When the user inputs an expression index that doesn't correspond to a slider, the expression can be overridden by the slider, screenshots & gif below for y=mx + b:

Screen Shot 2019-06-18 at 3 30 10 PM

capture burst, slider index set to 1, produces the following:

y_mx_b

My aim was to only allow the user to select from sliders that wont override expressions by filtering the expressions list from calculator.getExpressions() for only expressions whose latex follows the pattern of 1 variable being assigned a numerical value. I implemented a dropdown menu, with default values that throw errors when the users tries to capture an image burst. If there are no sliders, the defaul option is simply No Sliders.

Screen Shot 2019-06-18 at 3 36 18 PM

To do this, I needed to add `burstSlider` to the redux store in ui, which gets updated every time the burst panel is opened.

@ctlusto 
@alissarenz 
@mmmaaatttttt
Chris, please wait for Alissa to approve before review!

### CODE REVIEW

File: src/components/Burst.css [hunk 1]
```diff
@@ -31,6 +31,16 @@
...
 }
 
+.Burst-dropdown {
+  height: 33px;
+  width: 114px;
+  margin-bottom: 15px;
+  margin-top: 5px;
+  border: 1px solid #000;
+  font-size: 1em;
+  outline: none !important;
```
Comment (misscoded) [inline src/components/Burst.css:41]: What was the reason behind this (outline)?

File: src/actions/index.js [hunk 2]
```diff
@@ -217,3 +222,16 @@ export const generateGIF = (images, opts) => (dispatch, getState) => {
...
+
+export const getBurstSliders = () => dispatch => {
+  const sliders = getSliderExpressions();
+  if (!sliders.length) {
+    dispatch(flashError(noSlidersFound()));
+  }
+  dispatch(updateBurstSliders(sliders));
+};
+
+const updateBurstSliders = sliders => ({
```
Comment (misscoded) [inline src/actions/index.js]: In the rest of this file, the actions appear to be at the top. Is there a particular reason you've placed this one apart from the rest?

File: src/components/Burst.css (hunk 1 above)
Comment (japamat0) [reply src/components/Burst.css:41]: this should eliminate the blue outline that appears on focus to follow the same styling as other gifsmos inputs, e.g. `.Burst-input` and `.Burst-button` screenshots related:
Screen Shot 2019-06-21 at 11 42 43 AM
Screen Shot 2019-06-21 at 11 42 58 AM

File: src/actions/index.js (hunk 2 above)
Comment (japamat0) [reply src/actions/index.js]: good catch! moved from thunk section to action section

**Comment by mmmaaatttttt** [comment]
@japamat -- am i using this wrong or are there bugs in this PR? I tried pasting in the following:

https://www.desmos.com/calculator/gzmoe8w5hx

there are 3 sliders in this graph.

But when I try to select one of them from the dropdown, i either get a warning message about not defining a slider (for the second two), or a straight up error (for the first one). can you reproduce?

File: src/components/Burst.js [hunk 3]
```diff
@@ -38,24 +37,34 @@ class Burst extends Component {
...
           })}
-          type="number"
           name="idx"
           aria-label="slider index"
-          value={isNaN(idx) ? '' : idx}
           onChange={this.handleInputUpdate}
-        />
+        >
+          <option value={null} defaultValue>
+            {burstSliders.length ? 'Choose Slider' : 'No Sliders'}
```
Comment (mmmaaatttttt) [inline src/components/Burst.js]: can you change the 'Choose Slider' text to something shorter? the text is getting cut off:

Screen Shot 2019-06-25 at 11 38 27 AM

File: src/components/Burst.js [hunk 4]
```diff
@@ -38,24 +37,34 @@ class Burst extends Component {
...
+          </option>
+          {burstSliders.map(exp => {
+            return (
+              <option key={`slider-${exp.id}`} value={exp.id}>
+                {' '}
```
Comment (mmmaaatttttt) [inline src/components/Burst.js]: do you need this?

**Review by mmmaaatttttt — CHANGES_REQUESTED** [review]
@japamat - nothing in here looks problematic, but i think there are some :bug: in here. happy to chat more when you're free.

**Comment by japamat0** [comment]
@mmmaaatttttt 
I fixed the bugs you caught, let me know if you are still getting those

File: src/components/Burst.js (hunk 4 above)
Comment (japamat0) [reply src/components/Burst.js]: Removed

File: src/components/Burst.js (hunk 3 above)
Comment (japamat0) [reply src/components/Burst.js]: Changed to `Pick Slider`, now it fits

**Pull request closed without merging**